### PR TITLE
remove check_storage_aliases

### PR DESCRIPTION
**Pull request by jjsjann123**
The motivation is to remove the use of `numel` in code logic, since number of elements doesn't play well with dynamic shape yet.

I don't know what `check_storage_aliases` is protecting us against. So I removed the whole thing and looks like we don't have any test validating the logic neither.
This is a follow up on conversation here: https://github.com/Lightning-AI/lightning-thunder/pull/877/files#r1844732984
cc'ing @crcrpar for visibility.

### CODE REVIEW

File: thunder/__init__.py [hunk 1]
```diff
@@ -575,15 +575,13 @@ def get_computation_and_inputs(*args, **kwargs):
...
-                    tensor_args_consumed_by_inplace_grouped_by_numel: dict[int, list[TensorProxy]] = defaultdict(list)
                     for bsym in filter(lambda b: b.sym.id == prims.PrimIDs.COPY_, computation_trc.bound_symbols):
                         t = bsym.flat_proxy_args[1]
                         index = arg_to_idx[t]
-                        numel = t.numel
-                        tensor_args_consumed_by_inplace_grouped_by_numel[numel].append(index)
                         tensor_indices.append(index)
-                    if len(tensor_args_consumed_by_inplace_grouped_by_numel) > 1:
-                        vanilla_tensor_args = set(tensor_indices)
+                    tmp_vanilla_tensor_args = set(tensor_indices)
```
Comment (mruberry) [inline thunder/__init__.py:582]: Comments please! This file is getting hard to read

Comment (jjsjann123) [reply thunder/__init__.py:582]: oh sorry, this is just a smoke test. I should have added a do not review on the title first. :bow:

**Comment by jjsjann123** [comment]
failing tests seems to be coming from nvfuser.

File: thunder/__init__.py [hunk 2]
```diff
@@ -564,27 +560,6 @@ def get_computation_and_inputs(*args, **kwargs):
...
-
-                    tensor_indices: int = []
-                    tensor_args_consumed_by_inplace_grouped_by_numel: dict[int, list[TensorProxy]] = defaultdict(list)
-                    for bsym in filter(lambda b: b.sym.id == prims.PrimIDs.COPY_, computation_trc.bound_symbols):
-                        t = bsym.flat_proxy_args[1]
-                        index = arg_to_idx[t]
-                        numel = t.numel
-                        tensor_args_consumed_by_inplace_grouped_by_numel[numel].append(index)
-                        tensor_indices.append(index)
-                    if len(tensor_args_consumed_by_inplace_grouped_by_numel) > 1:
```
Comment (jjsjann123) [inline thunder/__init__.py:586]: I don't see the logic behind this conditional. i.e. number of groups of tensor args with the same number of elements looks odd here.

File: thunder/__init__.py [hunk 3]
```diff
@@ -823,8 +785,6 @@ def fn_(*args, **kwargs) -> Any:
 
         cache_entry, inps, pro_to_epi = get_computation_and_inputs(*args, **kwargs)
 
-        check_storage_aliases(cache_entry, inps)
```
Comment (jjsjann123) [inline thunder/__init__.py:826]: if we do not have a test case to validate this, I'm not sure what this check is trying to do.

**Comment by mruberry** [comment]
@jjsjann123 Is this ready for a review now?

**Comment by jjsjann123** [comment]
> @jjsjann123 Is this ready for a review now?

Yes. Since this is only removing code `+0 -40`. I didn't ended up putting any comment in.

**Review by mruberry — APPROVED** [review]

**Pull request merged**